scripts/update_changelog.py

```diff
@@ -77,12 +100,59 @@ def generate_md(grouped_pulls: dict[str, list[github.PullRequest.PullRequest]])
     """Generate markdown file from Jinja template."""
     changelog_template = ROOT / ".github" / "changelog-template.md"
     template = Template(changelog_template.read_text(), autoescape=True)
-    return template.render(merge_date=MERGED_DATE, grouped_pulls=grouped_pulls)
+    return template.render(grouped_pulls=grouped_pulls)
+
+
+def write_changelog(file_path: Path, release: str, content: str) -> None:
+    """Write Release details to the changelog file."""
+    content = f"## {release}\n{content}"
+    old_content = file_path.read_text()
+    updated_content = old_content.replace(
+        "<!-- GENERATOR_PLACEHOLDER -->",
+        f"<!-- GENERATOR_PLACEHOLDER -->\n\n{content}",
+    )
+    file_path.write_text(updated_content)
+
+
+def update_version(file_path: Path, release: str) -> None:
+    """Update template version in setup.py."""
+    old_content = file_path.read_text()
+    updated_content = re.sub(
+        r'\nversion = "\d+\.\d+\.\d+"\n',
+        f'\nversion = "{release}"\n',
+        old_content,
+    )
+    file_path.write_text(updated_content)
+
+
+def update_git_repo(paths: list[Path], release: str) -> None:
+    """Commit, tag changes in git repo and push to origin."""
+    repo = git.Repo(ROOT)
+    for path in paths:
+        repo.git.add(path)
+    message = f"Release {release}"
+
+    user = repo.git.config("--get", "user.name")
+    email = repo.git.config("--get", "user.email")
+
+    repo.git.commit(
+        m=message,
+        author=f"{user} <{email}>",
+    )
+    repo.git.tag("-a", release, m=message)
+    server = f"https://{GITHUB_TOKEN}@github.com/{GITHUB_REPO}.git"
+    print(f"Pushing changes to {GIT_BRANCH} branch of {GITHUB_REPO}")
+    repo.git.push(server, GIT_BRANCH)
+    repo.git.push("--tags", server, GIT_BRANCH)
```

Comment:
That function has been heavily inspired by [how `python-semantic-release` does it](https://github.com/relekang/python-semantic-release/blob/master/semantic_release/vcs_helpers.py).